Index: h5py/setup_build.py
===================================================================
--- h5py.orig/setup_build.py	2025-02-22 14:47:56.523752641 +0100
+++ h5py/setup_build.py	2025-02-22 14:49:24.480563042 +0100
@@ -18,6 +18,8 @@
 import api_gen
 from setup_configure import BuildConfig
 
+# set flavour to 'pre_h5py_suf' to enable alternative builds e.g. _debian_h5py_mpi
+flavour=''
 
 def localpath(*args):
     return op.abspath(op.join(op.dirname(__file__), *args))
@@ -116,9 +118,13 @@
             #settings['runtime_library_dirs'] = settings['library_dirs']
 
         def make_extension(module):
-            sources = [localpath('h5py', module + '.pyx')] + EXTRA_SRC.get(module, [])
+            sources = [localpath('h5py', flavour, module+'.pyx')] + EXTRA_SRC.get(module, [])
             settings['libraries'] += EXTRA_LIBRARIES.get(module, [])
-            return Extension('h5py.' + module, sources, **settings)
+            if flavour:
+                h5py_module='h5py'+'.'+flavour+'.'+module
+            else:
+                h5py_module='h5py'+'.'+module
+            return Extension(h5py_module, sources, **settings)
 
         return [make_extension(m) for m in MODULES]
 
@@ -148,11 +154,11 @@
                 f"{config.hdf5_version} from environment variable or library)"
             )
 
-        config_file = localpath('h5py', 'config.pxi')
+        config_file = localpath('h5py', flavour, 'config.pxi')
 
         # Refresh low-level defs if missing or stale
         print("Executing api_gen rebuild of defs")
-        api_gen.run()
+        api_gen.run(flavour=flavour)
 
         # Rewrite config.pxi file if needed
         s = f"""\
Index: h5py/api_gen.py
===================================================================
--- h5py.orig/api_gen.py	2025-02-22 14:47:56.523752641 +0100
+++ h5py/api_gen.py	2025-02-22 14:50:58.621428833 +0100
@@ -189,15 +189,15 @@
     def __init__(self, config) -> None:
         self.config = config
 
-    def run(self):
+    def run(self, flavour=''):
 
         # Function definitions file
-        self.functions = Path('h5py', 'api_functions.txt').open('r')
+        self.functions = Path('h5py', flavour, 'api_functions.txt').open('r')
 
         # Create output files
-        path_raw_defs = Path('h5py', '_hdf5.pxd.new')
-        path_cython_defs = Path('h5py', 'defs.pxd.new')
-        path_cython_imp = Path('h5py', 'defs.pyx.new')
+        path_raw_defs = Path('h5py', flavour, '_hdf5.pxd.new')
+        path_cython_defs = Path('h5py', flavour, 'defs.pxd.new')
+        path_cython_imp = Path('h5py', flavour, 'defs.pyx.new')
         self.raw_defs = path_raw_defs.open('w')
         self.cython_defs = path_cython_defs.open('w')
         self.cython_imp = path_cython_imp.open('w')
@@ -319,11 +319,11 @@
         self.cython_imp.write(imp)
 
 
-def run():
+def run(flavour=''):
     # Get configuration from environment variables
     config = BuildConfig.from_env()
     lp = LineProcessor(config)
-    lp.run()
+    lp.run(flavour=flavour)
 
 
 if __name__ == '__main__':
