Index: h5py/setup.py
===================================================================
--- h5py.orig/setup.py	2021-03-23 00:27:10.695445250 +0100
+++ h5py/setup.py	2021-03-23 00:27:10.691445225 +0100
@@ -49,7 +49,7 @@
     "Cython >=0.29.14; python_version=='3.8'",
     "Cython >=0.29.15; python_version>='3.9'",
 ] + [
-    f"numpy =={np_min}; python_version{py_condition}"
+    f"numpy >={np_min}; python_version{py_condition}"
     for np_min, py_condition in NUMPY_MIN_VERSIONS
 ]
 
