Index: h5py/h5py/tests/test_file2.py
===================================================================
--- h5py.orig/h5py/tests/test_file2.py	2024-12-05 17:29:35.923904313 +0100
+++ h5py/h5py/tests/test_file2.py	2024-12-05 17:29:35.919904288 +0100
@@ -21,6 +21,8 @@
 import io
 import tempfile
 import os
+import sysconfig
+import sys
 
 
 def nfiles():
@@ -212,6 +214,7 @@
         f.create_dataset('test', data=list(range(12)))
         self.assertRaises(Exception, list, f['test'])
 
+    @ut.skipIf(sys.version_info.major==3 and sysconfig.get_config_var("Py_DEBUG"), 'test fails in python3-dbg, see https://github.com/h5py/h5py/issues/1291')
     def test_exception_write(self):
 
         class BrokenBytesIO(io.BytesIO):
